enterprise/server/remote_execution/dirtools/dirtools_test.go

```diff
@@ -10,18 +11,322 @@ import (
+
+		expectedResult *repb.ActionResult
+		expectedInfo   *dirtools.TransferInfo
+	}{
+		{
+			name:           "NoFiles",
+			cmd:            &repb.Command{},
+			directoryPaths: []string{},
+			fileContents:   map[string]string{},
+			symlinkPaths:   map[string]string{},
+			expectedResult: &repb.ActionResult{},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        0,
+				BytesTransferred: 0,
+			},
+		},
+		{
+			name: "SomeFile",
+			cmd: &repb.Command{
+				OutputFiles: []string{"fileA.txt"},
+			},
+			directoryPaths: []string{},
+			fileContents: map[string]string{
+				"fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{},
+			expectedResult: &repb.ActionResult{
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        1,
+				BytesTransferred: 1,
+			},
+		},
+		{
+			name: "FileAndDirectory",
+			cmd: &repb.Command{
+				OutputFiles:       []string{"a/fileA.txt"},
+				OutputDirectories: []string{"a"},
+			},
+			directoryPaths: []string{
+				"a",
+			},
+			fileContents: map[string]string{
+				"a/fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{},
+			expectedResult: &repb.ActionResult{
+				OutputDirectories: []*repb.OutputDirectory{
+					{
+						Path: "a",
+						TreeDigest: &repb.Digest{
+							SizeBytes: 85,
+							Hash:      "895545df6841b7efb2e9cc903a4eac7a60c645199be059f6056817ae6feb071d",
+						},
+					},
+				},
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "a/fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        2,
+				BytesTransferred: 84,
+			},
+		},
+		{
+			name: "SymlinkToFile",
+			cmd: &repb.Command{
+				OutputFiles: []string{
+					"fileA.txt",
+					"linkA.txt",
+				},
+			},
+			directoryPaths: []string{},
+			fileContents: map[string]string{
+				"fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{
+				"linkA.txt": "fileA.txt",
+			},
+			expectedResult: &repb.ActionResult{
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+				OutputFileSymlinks: []*repb.OutputSymlink{
+					{
+						Path:   "linkA.txt",
+						Target: "fileA.txt",
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        1,
+				BytesTransferred: 1,
+			},
+		},
+		{
+			name: "SymlinkToDirectory",
+			cmd: &repb.Command{
+				OutputFiles: []string{"a/fileA.txt"},
+				OutputDirectories: []string{
+					"a",
```

Comment:
ditto here - I think in practice we wouldn't see `a/fileA.txt` nested under an output directory. (there are also a few more occurrences of this below)